### PR TITLE
Adding a readiness probe to kubectl e2e tests.

### PR DESCRIPTION
**Pull request by maisem**
ref #20268

cc @lavalamp @ncdc @bprashanth

### CODE REVIEW

File: examples/readiness-probe-pod [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copy of pod.yaml without file extension for test
```
Comment (lavalamp) [inline examples/readiness-probe-pod]: Move under the e2e test data directory if you're copying it? Also, please add a .yaml file extension.

Comment (bprashanth) [inline examples/readiness-probe-pod]: Yeah this is why I'd suggested just adding the probe inline. If you move to test/ you'll need to make sure the directory you put it in is included in the test tarball. @ixdy where would be a safe place to put yaml? IIRC the last time I shoved something in test/e2e/testing-manifests it wasn't available on the server, and I ended up just inling it anyway

Comment (ncdc) [inline examples/readiness-probe-pod]: Why not modify docs/user-guide/pod.yaml to add the probe?

Comment (bprashanth) [inline examples/readiness-probe-pod]: I slightly prefer keeping the example simple, but I'm not really against it. Just make sure you expalin what it is, if there's an accompanying example doc.

Comment (ixdy) [inline examples/readiness-probe-pod]: `test/images/network-tester/rc.json` and `test/images/network-tester/service.json` are [explicitly included in the test tarball](https://github.com/kubernetes/kubernetes/blob/master/hack/lib/golang.sh#L90-L91). We should probably do something similar for `test/e2e/testing-manifests`.

Comment (ixdy) [inline examples/readiness-probe-pod]: (note that I can't find any reference to those two files being used in any e2e tests anymore, so we probably don't need those lines...)

Comment (ixdy) [inline examples/readiness-probe-pod]: Also, all of `test/images` is included in a separate place: https://github.com/kubernetes/kubernetes/blob/master/build/common.sh#L943-L946

(We probably should not be shipping all of `test/images` in the test tarball...)

Comment (bgrant0607) [inline examples/readiness-probe-pod]: Please don't put this here. Why not in test/e2e?

Comment (bgrant0607) [inline examples/readiness-probe-pod]: Ok, I see the comment above, but it really doesn't belong here.

Comment (bprashanth) [inline examples/readiness-probe-pod]: Recommend just adding testing-manifests to the tarball as recommended by ixdy, moving the stuff already in there into a test specific subdir (testing-manifests/serviceloadbalancer/), and putting this into testing-manifests/kubectl/

**Comment by maisem** [comment]
@bprashanth - ping

**Comment by bprashanth** [comment]
Thanks, LGTM

**Comment by lavalamp** [comment]
@maisem you should tag @k8s-oncall to do manual merges, not merge yourself. Depending on what the build cop is doing at the moment, it could be very inconvenient to them to have a surprise merge.

**Comment by maisem** [comment]
sorry, my bad. I didn't realize that was the process, should I revert it?

**Comment by mikedanese** [comment]
Just let merge-bot do it's thing and process the queue. If you need cut the queue and get something merged pronto, ask build cop oncall to do it for you since they are watching the build. No need to revert this.

**Comment by maisem** [comment]
Ah ok, that makes sense. I just hit the shiny green button :)
I'll keep that in mind. Thanks!

**Comment by mikedanese** [comment]
docs kind of https://github.com/kubernetes/kubernetes/blob/master/docs/devel/on-call-build-cop.md#build-copping

**Pull request merged**